components/remote_settings/android/src/test/java/org/mozilla/appservices/remotesettings/RemoteSettingsTest.kt

```diff
@@ -0,0 +1,149 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.appservices.remotesettings
```

Comment:
It was frustrating to keep track of while dealing with tooling, so I think this nit has merit anyway. I only did it because I had been under the impression that neither underscores or hyphens were valid characters in package names. This comment made me look into it though, it turns out that [Oracle suggests using underscores and that hyphens are illegal characters](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html). Without more research, I would guess the hyphens in your example are legal because that is a reference to a plugin and not a package.

Reply:
I did try this, but am now running into Android build issues. I'm not sure they're related yet, because reverting didn't seem to fix them though I may have missed a step in the revert. FWIW, the issue is:

`A problem occurred configuring project ':nimbus-gradle-plugin'.`

Stacktrace top is:
`Cannot get property 'build' on extra properties extension as it does not exist`

Reply:
Still working through these, but it seems like the linter won't allow package names with hyphens anyway. I'm of the opinion that we should still keep snake-cased crate names for legibility, and combine the words in Android package names to avoid these issues.

Reply:
Deleted a couple old comments but TL;DR: the complexity analysis tool we use `detekt`, won't allow hyphenated names. I'd prefer to just switch the naming convention here for Android's sake instead of switching all the Rust code to use concatenated names, but let me know if you'd prefer that solution.

Reply:
I agree 100% - but...

> fxaclient project => fxa-client crate, errorsupport => error-support, syncmanager => sync_manager

We'd probably still want some consistency there, right? ie, IIUC, in all the above cases, the crate names would *always* have underscores, but the package names would be inconsistent between dashes and underscores - which is already true today. Further, there's also the directory name to consider :)

Personally, I think I'd vote for "use dashes everywhere they are allowed" - so we'd have a package and directory with name "sync-manager" while the crate name would be "sync_manager". But this is getting off-topic for this PR - I think what's here is great (and it is all that much cleaner now we've killed that extra crate!)

Reply:
> IIUC, in all the above cases, the crate names would always have underscores, but the package names would be inconsistent between dashes and underscores - which is already true today.

Hmmm, am I looking in the wrong place? 😅 I see mostly concatenated (`oneword`) names, a couple of hyphens, and no underscores in the package or project names in `.buildconfig-android.yml`—the only underscores there are in the directory names—and most of the package names in `AndroidManifest.xml` are concatenated, with a couple of underscores.

> Further, there's also the directory name to consider :)

Ohh, I was using "directory name" and "crate name" interchangeably, and didn't even think about them being different! Except for the `support/` components, I don't see other examples under `components/` where the directory name is different from the crate name (in `Cargo.toml`).

Reply:
>  package or project names in .buildconfig-android.yml

Right! I was talking just about the rust package names - so many places to get it wrong :)